### PR TITLE
Changes to Common Controls Baseline

### LINKED ISSUE #240: GWS.COMMONCONTROLS.14.1v0.1 implementation steps incomplete
The baseline requirement:
image

What the implementation steps say to do:
image

This will send logs to GCP. But the requirement is to send the logs to "the agency's centralized SEIM," which seems like would require an additional step, depending on what the agency is using as their SEIM.

### LINKED ISSUE #252: Consider increasing super admin limit
Per https://github.com/cisagov/ScubaGoggles/pull/242#discussion_r1548726073, "Note that there is an equivalent policy in the Microsoft 365 AAD/Entra ID SCuBA Baseline. The maximum was bumped up to 8 users from 4 because of pilot feedback from over there."

### LINKED ISSUE #274: What is the Guidance for the "Manage access to services that aren't controlled individually" Setting?
# 💡 Summary #
What is the Guidance for the "Manage access to services that aren't controlled individually" Setting?

Manage access to services that aren't controlled individually
You can manage access to Google services that aren't shown in the Admin console, and therefore, don't have an individual ON/OFF control. 

https://support.google.com/a/answer/7646040?hl=en#:~:text=For%20services%20that%20don't,or%20just%20specific%20organizational%20units.

### Reasons to restrict access to services ###
Some Google services don't have their own On or Off setting in the Admin console, and Google may release new services at any time. You might want to restrict users from accessing services with their managed Google Account for reasons such as:

- Company policy—Your organization might need control over the services your users can use with their organization-managed Google Accounts.
- Industry regulations—Your organization might be subject to industry regulations that require administrative control over users' data.
- School restrictions—Your educational institution might need to restrict use of services for policy reasons or because it doesn't have parental approval to provide them to students.

### How restricting access to services affects users ###

- If you turn off Google services that you don't control individually in the Admin console, users can't use services that might be helpful to them. If users are already using these services for work purposes, restricting access to them might disrupt their work.
- If users try to sign in to a restricted service, they're usually redirected to a page that tells them the service is unavailable and why.

### Turn services on or off ###
[Sign in](https://admin.google.com/) to your [Google Admin console](https://support.google.com/a/answer/182076).
Sign in using an administrator account, not your current account anthony.benjamen.lutz@gmail.com

1. In the Admin console, go to Menu and then Appsand thenAdditional Google services.
2. In the message at the top of the page, click Change.
3. Choose how you want to restrict services and follow the steps:

- Turn services on or off for all users in your organization
- In the left panel, make sure Settings for all organizational units is selected.
- Choose to turn services on or off for all users.
- Click Save.

4. Turn services on or off for users in a specific organization unit

- In the left panel, select the organizational unit.
- Choose to turn services on or off for users.
- Click Override to keep this setting the same, even if the parent setting changes.
- If the organizational unit's status is already Overridden, choose an option:
- Inherit—Reverts to the same setting as its parent.
- Save—Saves your new setting (even if the parent setting changes).

It can take up to 48 hours for the new setting to take effect.

Tip: If you turn off services, consider communicating your organization's policy to users.

Please provide details for implementation, such as:

- an example for how this would be used
- what this would look like
- how this would act
- any related work, including links to related issues

## Acceptance criteria ##

Yes let us know the recommended setting.

### LINKED ISSUE #276: Google Product Update for Multi-Party Approval
Google released a feature where multiple super admins need to approve any changes to the settings listed below:
- 2 Step Verification
- Account Recovery
- Advanced Protection
- Google Session Control
- Login Challenges
- Passwordless (beta)

### PR DESCRIPTION
**Pull request by jkaufman-mitre**
## 🗣 Description ##

The following changes were made within the common controls baseline:
- Policy 2.2 was removed
- Policy 10.1 was removed
- Policy Group 16, Additional Google Services, was created
- Policy group 17, Multi-Party Approval, was created
- Policy 14.1 implementation was updated.
- Change 11.1 to SHALL

### 💭 Motivation and context

Fixes #240
Fixes #252
Fixes #274 
Fixes #276 

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [ ] Changes are limited to a single goal - *eschew scope creep!*
- [ ] If applicable, *All* future TODOs are captured in issues, which are referenced in the PR description.
- [x] The relevant issues PR resolves are linked preferably via [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [x] All relevant type-of-change labels have been added.
- [x] I have read and agree to the [CONTRIBUTING.md](https://github.com/cisagov/ScubaGoggles/blob/main/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge Checklist

- [ ] This PR has been smoke tested to ensure main is in a functional state when this PR is merged.
- [ ] Squash all commits into one PR level commit using the `Squash and merge` button.

## ✅ Post-merge Checklist

- [ ] Delete the branch to clean up.
- [ ] Close issues resolved by this PR if the [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) did not activate.

### CODE REVIEW

**Review by adhilto — CHANGES_REQUESTED** [review]
GWS.COMMONCONTROLS.2.1v0.2 still needs more work.

The note, "granular controls may be used if the agency needs it," is misleading because it kind of implies that specifying what controls you want context aware access to apply is optional. It's not. Simply doing what is currently the implementation for 2.1 doesn't do anything. The full instructions need to include the following elements:

1) Turn on context aware access (as instructed in the current implementation, see screenshot) 
image

2) Create one or more access levels, as needed
image
image

3) Determine the conditions of the rule, per agency discretion.
image

4) Assign the access levels
image

Finally, there are still a lot of open questions left.
- What should the conditions of the access levels be? Entirely up to agency discretion?
- Which apps should be assigned to those levels? All? Just the admin center? Agency discretion?
- Which users should be assigned to those levels? They can be assigned at the OU and or group level. All? Or agency discretion?
- What mode should those assignments be in? Just monitor? Or active? Left to agency discretion?

Recommendations:
- Clarify that "Note:" for 2.1. Maybe: "Note: the implementation details of context-aware access use cases will vary per agency. Refer to [Google's documentation](https://support.google.com/a/answer/12643733) on implementing context-aware access for your specific use cases."
- Augment the implementation steps to include all that I detailed above.
- Delete the implementation steps for 2.2 (they're still there).
- Address the open questions listed above.

File: baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md [hunk 1]
```diff
@@ -1139,3 +1130,38 @@ To configure Supplemental Data Storage per the policy:
...
+
+This section covers the Google services that do not have an individual control and whether these services are on or off.
+
+### Policy
+
+#### GWS.COMMONCONTROLS.16.1v0.1
+Service status SHOULD be set to OFF for everyone.
+
+- _Rationale_: Allowing access to additional google services without a need creates unnceccessary vulnerabilities within the Google Workspace enviroment. By turning these services off, it mitigates the risk by not allowing access.
+- _Last modified:_ October 30, 2023
```
Comment (adhilto) [inline baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md:1145]: Is this date accurate?

File: baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md [hunk 2]
```diff
@@ -1139,3 +1130,38 @@ To configure Supplemental Data Storage per the policy:
...
+
+- MITRE ATT&CK TTP Mapping
+  - Pending
```
Comment (adhilto) [inline baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md:1148]: Should be complete before merging.

File: baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md [hunk 3]
```diff
@@ -1139,3 +1130,38 @@ To configure Supplemental Data Storage per the policy:
...
+
+### Resources
+-	[GWS Admin Help \| Data regions: Choose a geographic location for your data](https://support.google.com/a/answer/7630496)
+-	[GWS Admin Help \| What data is covered by a data region policy?](https://support.google.com/a/answer/9223653)
```
Comment (adhilto) [inline baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md:1152]: These look like the wrong resources.

File: baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md [hunk 4]
```diff
@@ -1139,3 +1130,38 @@ To configure Supplemental Data Storage per the policy:
...
+### Implementation
+
+#### GWS.COMMONCONTROLS.16.1v0.1 Instructions
+To configure Data Regions per the policy:
+1.	Sign in to the [Google Admin console](https://admin.google.com) as an administrator.
+2.	Navigate to **Account** -> **Account settings**.
+3.	Click the **Data Regions** card.
+4.	Click the **Data Regions** policy card.
+5.	Select the radio button option: "**United States**"
+6.	Click **Save**.
```
Comment (adhilto) [inline baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md:1167]: These are the wrong steps.

File: baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md [hunk 5]
```diff
@@ -1139,3 +1130,38 @@ To configure Supplemental Data Storage per the policy:
...
+## 16. Additional Google Services
+
+This section covers the Google services that do not have an individual control and whether these services are on or off.
+
+### Policy
+
+#### GWS.COMMONCONTROLS.16.1v0.1
+Service status SHOULD be set to OFF for everyone.
+
+- _Rationale_: Allowing access to additional google services without a need creates unnceccessary vulnerabilities within the Google Workspace enviroment. By turning these services off, it mitigates the risk by not allowing access.
```
Comment (adhilto) [inline baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md:1144]: - "creates vulnerabilities" is too confident. It seems to imply that other Google services are vulnerable, which may be a possibility, but is far from certain. "May create" might be better.
- Two spelling errors
```suggestion
- _Rationale_: Allowing access to additional google services without a need may create unnecessary vulnerabilities within the Google Workspace environment. By turning these services off, it mitigates the risk by not allowing access.
```

File: baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md [hunk 6]
```diff
@@ -1139,3 +1130,38 @@ To configure Supplemental Data Storage per the policy:
...
+
+
+## 16. Additional Google Services
+
+This section covers the Google services that do not have an individual control and whether these services are on or off.
+
+### Policy
+
+#### GWS.COMMONCONTROLS.16.1v0.1
+Service status SHOULD be set to OFF for everyone.
```
Comment (adhilto) [inline baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md:1142]: Policy is too vague. Service status for what?
```suggestion
Service status for Google services that do not have an individual control SHOULD be set to OFF for everyone.
```

File: baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md [hunk 7]
```diff
@@ -1139,3 +1130,38 @@ To configure Supplemental Data Storage per the policy:
 3.	Click the **Supplemental Data Storage** card.
 4.	Ensure the checkbox for "**Russian Federation**" is unchecked.
 6.	Click **Save**.
+
+
+## 16. Additional Google Services
+
+This section covers the Google services that do not have an individual control and whether these services are on or off.
```
Comment (adhilto) [inline baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md:1137]: Do we know which services specifically would be affected by this policy? I looked at the linked resources but couldn't find a concrete list.

File: baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md [hunk 8]
```diff
@@ -1139,3 +1130,38 @@ To configure Supplemental Data Storage per the policy:
...
+-	[GWS Admin Help \| Data regions: Choose a geographic location for your data](https://support.google.com/a/answer/7630496)
+-	[GWS Admin Help \| What data is covered by a data region policy?](https://support.google.com/a/answer/9223653)
+
+### Prerequisites
+
+- Super Admin role
+
+### Implementation
+
+#### GWS.COMMONCONTROLS.16.1v0.1 Instructions
```
Comment (adhilto) [inline baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md:1160]:
```suggestion
#### GWS.COMMONCONTROLS.16.1v0.2 Instructions
```

File: baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md [hunk 9]
```diff
@@ -1139,3 +1130,38 @@ To configure Supplemental Data Storage per the policy:
...
 6.	Click **Save**.
+
+
+## 16. Additional Google Services
+
+This section covers the Google services that do not have an individual control and whether these services are on or off.
+
+### Policy
+
+#### GWS.COMMONCONTROLS.16.1v0.1
```
Comment (adhilto) [inline baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md:1141]:
```suggestion
#### GWS.COMMONCONTROLS.16.1v0.2
```

**Review by adhilto — CHANGES_REQUESTED** [review]
Feedback for GWS.COMMONCONTROLS.16.1

**Comment by jkaufman-mitre** [comment]
Removed Policy 10.1 because there is no implementation for the policy. Agencies will be evaluated on their in-house procedures for items such as this as part of their NIST 800-53 control assessments.

File: baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md (hunk 1 above)
Comment (jkaufman-mitre) [reply baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md]: Fixed the date

File: baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md (hunk 2 above)
Comment (jkaufman-mitre) [reply baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md:1131]: They are being worked on.

File: baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md (hunk 4 above)
Comment (mdueltgen) [reply baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md]: Fixed the implementation steps

File: baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md (hunk 7 above)
Comment (mdueltgen) [reply baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md]: https://support.google.com/a/answer/7646040?fl=1&sjid=6354300702774302500-NA
https://support.google.com/a/answer/181865?hl=en&fl=1&sjid=6354300702774302500-NA#zippy=%2Cturn-services-on-or-off-for-users
I can't find a concrete list either, is it something we think should be included?

**Comment by jkaufman-mitre** [comment]
Moved 11.2 to Policy Group 10. Fixes #318

File: baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md (hunk 7 above)
Comment (adhilto) [reply baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md]: Ideally yes, so people can understand the implications of turning these services off. But if Google hasn't published that information, not much we can do.

File: baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md [hunk 10]
```diff
@@ -1139,3 +1113,67 @@ To configure Supplemental Data Storage per the policy:
...
+To configure additional services per the policy:
+1.	Sign in to the [Google Admin console](https://admin.google.com) as an administrator.
+2.	Navigate to **Apps** -> **Additional Google services**.
+3.	Click **CHANGE** at the top where it says if **Access to additional services without individual control for all organizational units is On/Off**.
+4.	Select the option: "**OFF for everyone**"
+6.	Click **Save**.
+
+## 17. Multi-Party Approval
+
+This section covers the oogle services that do not have an individual control and whether these serviGces are on or off.
```
Comment (adhilto) [inline baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md:1152]: Typo
```suggestion
This section covers the Google services that do not have an individual control and whether these services are on or off.
```

File: baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md [hunk 11]
```diff
@@ -1139,3 +1113,67 @@ To configure Supplemental Data Storage per the policy:
 3.	Click the **Supplemental Data Storage** card.
 4.	Ensure the checkbox for "**Russian Federation**" is unchecked.
 6.	Click **Save**.
+
+
+## 16. Additional Google Services
+
+This section covers the oogle services that do not have an individual control and whether these serviGces are on or off.
```
Comment (adhilto) [inline baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md:1120]: Typo
```suggestion
This section covers the Google services that do not have an individual control and whether these services are on or off.
```

File: baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md [hunk 12]
```diff
@@ -1139,3 +1113,67 @@ To configure Supplemental Data Storage per the policy:
...
+### Policy
+
+#### GWS.COMMONCONTROLS.17.1v0.2
+Require multi party approval for sensitive admin actions SHALL be enabled.
+
+- _Rationale_: Changes to certain admin settings creates the potential for security weaknesses. Therefore, requiring multiple super admins to approve changes to those settings mitigates the risk.
+- _Last modified:_ June 20, 2024
+
+- MITRE ATT&CK TTP Mapping
+  - Pending
```
Comment (adhilto) [inline baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md:1163]: Needs to be complete before merging.

File: baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md [hunk 13]
```diff
@@ -24,6 +24,7 @@ This baseline is based on Google documentation and addresses the following:
 - [System-Defined Rules](#13-system-defined-rules)
 - [Google Workspace Logs](#14-google-workspace-logs)
 - [Data Regions](#15-data-regions-and-storage)
+- [Additional Google Services](#16-additional-google-services)
```
Comment (adhilto) [inline baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md:27]: Need to add entry for 17.1

File: baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md [hunk 14]
```diff
@@ -1139,3 +1113,67 @@ To configure Supplemental Data Storage per the policy:
...
+## 17. Multi-Party Approval
+
+This section covers the oogle services that do not have an individual control and whether these serviGces are on or off.
+
+### Policy
+
+#### GWS.COMMONCONTROLS.17.1v0.2
+Require multi party approval for sensitive admin actions SHALL be enabled.
+
+- _Rationale_: Changes to certain admin settings creates the potential for security weaknesses. Therefore, requiring multiple super admins to approve changes to those settings mitigates the risk.
```
Comment (adhilto) [inline baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md:1159]: "Therefore" implies that the second sentence will logically follow from the first, which it doesn't. Recommend re-working. Additionally, including an example of what's included in "certain admin settings" would make this rationale much more convincing.

```suggestion
- _Rationale_: Changes to sensitive admin settings, such as disabling 2-step verification, could introduce serious vulnerabilities in the GWS environment. Requiring multiple super admins to approve changes to those settings mitigates the risk changing these settings pose.
```

**Comment by adhilto** [comment]
> Moved 11.2 to Policy Group 10. Fixes #318

Looks like this change might not have been pushed yet?

File: baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md (hunk 12 above)
Comment (jkaufman-mitre) [reply baselines/Common Controls Minimum Viable Secure Configuration Baseline v0.2.md:1163]: Working on these now.

**Comment by jkaufman-mitre** [comment]
Made Common Controls 11.1 a SHALL.

**Comment by jkaufman-mitre** [comment]
TTP Mappings have been added.

**Comment by jkaufman-mitre** [comment]
@adhilto @buidav Can you review this. This one is ready for review again.

**Comment by mdueltgen** [comment]
As discussed, removed Issue 290 for 2.2. Context Aware Access revamp will happen in the next release after Coast. 

@adhilto Please review the 2.1 section for Coast release including changes to the implementation steps.

**Comment by snarve** [comment]
11.2 is unchanged as @adhilto mentioned in the email thread. Since there is a separate issue for this( #318 ), recommend to create a separate branch for this and commit separately as this PR tackles multiple issues already. Having a separate branch and PR (per issue) would ease the tracking and updates.
@jkaufman-mitre @mdueltgen @adhilto @buidav any thoughts on this

**Comment by adhilto** [comment]
> 11.2 is unchanged as @adhilto mentioned in the email thread. Since there is a separate issue for this( #318 ), recommend to create a separate branch for this and commit separately as this PR tackles multiple issues already. Having a separate branch and PR (per issue) would ease the tracking and updates. @jkaufman-mitre @mdueltgen @adhilto @buidav any thoughts on this

I agree, that's the right call (with the caveat that branch be made after this one is merged in to ease merge conflicts).

**Comment by mdueltgen** [comment]
> > 11.2 is unchanged as @adhilto mentioned in the email thread. Since there is a separate issue for this( #318 ), recommend to create a separate branch for this and commit separately as this PR tackles multiple issues already. Having a separate branch and PR (per issue) would ease the tracking and updates. @jkaufman-mitre @mdueltgen @adhilto @buidav any thoughts on this
> 
> I agree, that's the right call (with the caveat that branch be made after this one is merged in to ease merge conflicts). I just edited the description of this PR to remove that issue so it's accurate and so that that issue won't auto-close once this PR is merged.

Sounds good I will make sure #318 and #290 are in different branches for the next release. I think now that those two have been removed from the description of this PR, I think we should be good to do review of PR as is and merge for Coast.

**Review by adhilto — APPROVED** [review]
Baseline looks good, and Rego changes are complete. @snarve did half yesterday, I did the last half today.

**Review by mdueltgen — APPROVED** [review]
Rev

**Pull request merged**